src/test/java/io/strimzi/kafka/bridge/http/TopicsTest.java

```diff
@@ -0,0 +1,518 @@
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.kafka.client.consumer.KafkaConsumer;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.bind.DatatypeConverter;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TopicsTest extends HttpBridgeTest{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TopicsTest.class);
+    // for periodic/multiple messages test
+    private static final int PERIODIC_MAX_MESSAGE = 10;
+    private static final int PERIODIC_DELAY = 200;
+    private static final int MULTIPLE_MAX_MESSAGE = 10;
+    private int count;
+
+    TopicService topicService(WebClient client) {
+        return new TopicService(client);
+    }
+
+    MultiMap defaultHeaders(JsonObject jsonObject) {
+        return new CaseInsensitiveHeaders()
+            .add("Content-length", String.valueOf(jsonObject.toBuffer().length()))
+            .add("Content-Type", BridgeContentType.KAFKA_JSON_JSON);
+    }
+
+
+    @Test
+    void sendMessageLackingRequiredProperty(VertxTestContext context) throws Throwable {
+        String topic = "sendMessageLackingRequiredProperty";
+        kafkaCluster.createTopic(topic, 1, 1);
+
+        String key = "my-key";
+
+        JsonArray records = new JsonArray();
+        JsonObject json = new JsonObject();
+        json.put("key", key);
+        records.add(json);
+
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.put("records", records);
+
+        Handler<AsyncResult<HttpResponse<JsonObject>>> handler =
+            ar ->
+                context.verify(() -> {
+                    assertTrue(ar.succeeded());
+                    HttpResponse<JsonObject> response = ar.result();
+                    HttpBridgeError error = HttpBridgeError.fromJson(response.body());
+                    assertEquals(HttpResponseStatus.BAD_REQUEST.code(), response.statusCode());
+                    assertEquals(HttpResponseStatus.BAD_REQUEST.code(), error.getCode());
+                    assertEquals("Validation error on: body.records[0] - $.records[0].value: is missing but it is required",
+                            error.getMessage());
+                });
+
+        topicService(client)
+            .headers(defaultHeaders(jsonObject))
+            .topicName(topic)
+            .jsonBody(jsonObject)
+            .sendRecord(context, handler);
+        context.completeNow();
```

Comment:
the context should not be completed here but in the handler as before. It's an async test.
As you did now, the `completeNow` will be called before the handler is executed.
Or am I missing something?

Reply:
You are right, we need to have it somewhere in a block of `context.verify()`